csharp/private/wrappers/dotnet.cc

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#ifdef _WIN32
+#include <errno.h>
+#include <process.h>
+#include <windows.h>
+#else  // not _WIN32
+#include <stdlib.h>
+#include <unistd.h>
+#endif  // _WIN32
+
+#include "tools/cpp/runfiles/runfiles.h"
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+std::string evprintf(std::string name, std::string path) {
+  std::stringstream ss;
+  ss << name << "=" << path;
+  return ss.str();
+}
+
+int main(int argc, char** argv) {
+  std::string error;
+
+  auto runfiles = Runfiles::Create(argv[0], &error);
+
+  if (runfiles == nullptr) {
+    std::cerr << "Couldn't load runfiles: " << error << std::endl;
+    return 101;
+  }
+
+  auto dotnet = runfiles->Rlocation("{DotnetExe}");
+  if (dotnet.empty()) {
+    std::cerr << "Couldn't find the .NET runtime" << std::endl;
+    return 404;
+  }
+
+  // Get the name of the directory containing dotnet.exe
+  auto dotnetDir = dotnet.substr(0, dotnet.find_last_of("/\\"));
+
+  /*
+  dotnet and nuget require these environment variables to be set
+  without them we cannot build/run anything with dotnet.
+
+  dotnet: HOME, DOTNET_CLI_HOME, APPDATA, PROGRAMFILES
+  nuget: TMP, TEMP, USERPROFILE
```

Comment:
We don't need to run `nuget` though?

Reply:
It isn't really about usage (as nothing in the resgen uses nuget), and more a matter of the `dotnet` executable establishing base assumptions. If we don't ensure those variables are accessible, then:

- `dotnet.exe` will reach out to `C:\WINDOWS\NuGetScratch\lock` to see if a lock is held on nuget
- `dotnet.exe` doesn't have permissions to access `C:\WINDOWS\Temp`, so access error
- Any error (including access) is seen as the lock being held already
- Since the lock is held, `dotnet.exe` errors, recommending you wait / release the lock

It may also look in the `AppData/Local` directory for the lock.

Reply:
Setting `--no-restore` and `--no-dependencies` may prevent the need for those environment variables.

Reply:
The nuget environment variables here were needed for the resx code, so I'm going to remove them from here so they can be included in the resx PR if necessary.

Reply:
Seems that `USERPROFILE` is needed for MacOS, so I will be adding that back. It looks like only the `TMP` variables were special for the resx code.